solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -45,10 +49,501 @@ describe("Bridge - Moving funds", () => {
+                    "0xbf198e8fff0f90af01024153701da99b9bc08dc5",
+                    "0xb2a89e53a4227dbe530a52a1c419040735fa636c",
+                  ]
+
+                  before(async () => {
+                    await createSnapshot()
+
+                    for (let i = 0; i < liveWallets.length; i++) {
+                      // eslint-disable-next-line no-await-in-loop
+                      await bridge.setWallet(liveWallets[i], {
+                        ...walletDraft,
+                        state: walletState.Live,
+                      })
+                    }
+
+                    await bridge.setLiveWalletsCount(liveWallets.length)
+                  })
+
+                  after(async () => {
+                    await restoreSnapshot()
+                  })
+
+                  context(
+                    "when the submitted target wallets count is same as the expected",
+                    () => {
+                      // The source wallet has a main UTXO with value of 26 BTC,
+                      // the max BTC transfer is 10 BTC by default (see
+                      // `constants.walletMaxBtcTransfer`) and the count of
+                      // live wallets is `5`. We compute the expected target
+                      // wallets count as:
+                      // `N = min(liveWalletsCount, ceil(walletBtcBalance / walletMaxBtcTransfer))`
+                      // so we have `N = min(5, 26 / 10) = min(5, 3) = 3`
+                      const expectedTargetWalletsCount = 3
+
+                      context(
+                        "when all target wallets are different than the source wallet",
+                        () => {
+                          context(
+                            "when all target wallets are in the Live state",
+                            () => {
+                              let tx: ContractTransaction
+
+                              const targetWallets = liveWallets.slice(
+                                0,
+                                expectedTargetWalletsCount
+                              )
+
+                              before(async () => {
+                                await createSnapshot()
+
+                                tx = await bridge
+                                  .connect(caller)
+                                  .submitMovingFundsCommitment(
+                                    ecdsaWalletTestData.pubKeyHash160,
+                                    mainUtxo,
+                                    walletMembersIDs,
+                                    walletMemberIndex,
+                                    targetWallets
+                                  )
+                              })
+
+                              after(async () => {
+                                await restoreSnapshot()
+                              })
+
+                              it("should store the target wallets commitment for given wallet", async () => {
```

Comment:
*the given

Reply:
https://github.com/keep-network/tbtc-v2/pull/208/commits/207daf09222830e9ff4805510006a59cc74f36ef